bundles/org.openhab.ui/web/src/pages/settings/persistence/persistence-edit.vue

```diff
@@ -0,0 +1,688 @@
+          </f7-col>
+          <f7-col>
+            <f7-list>
+              <f7-list-button v-if="isEditable" color="red" @click="deletePersistence">
+                Remove persistence configuration
+              </f7-list-button>
+            </f7-list>
+          </f7-col>
+        </f7-block>
+      </f7-tab>
+
+      <!-- Code Tab -->
+      <f7-tab id="code" @tab:show="() => { currentTab = 'code'; toYaml() }" :tab-active="currentTab === 'code'">
+        <f7-icon v-if="!isEditable" f7="lock" class="float-right margin"
+                 style="opacity:0.5; z-index: 4000; user-select: none;" size="50" color="gray"
+                 :tooltip="notEditableMgs" />
+        <editor v-if="currentTab === 'code'" class="persistence-code-editor"
+                mode="application/vnd.openhab.persistence+yaml" :value="persistenceYaml" @input="onEditorInput"
+                :read-only="!isEditable" />
+      </f7-tab>
+    </f7-tabs>
+  </f7-page>
+</template>
+
+<style lang="stylus">
+.moduleconfig-popup
+  .page-content
+    overflow-x hidden !important
+  .config-sheet, .parameter-group
+    margin-top 0 !important
+
+.modules
+  .swipeout-opened
+    .sortable-handler
+      display none
+  .item-media .icon
+    color var(--f7-theme-color)
+  .media-list
+    margin-bottom 0
+  .list
+    margin-top 0
+
+.persistence-code-editor.vue-codemirror
+  display block
+  top calc(var(--f7-navbar-height) + var(--f7-tabbar-height))
+  height calc(100% - 2 * var(--f7-navbar-height))
+  width 100%
+</style>
+
+<script>
+import DirtyMixin from '../dirty-mixin'
+import YAML from 'yaml'
+import CronStrategyPopup from '@/pages/settings/persistence/cron-strategy-popup.vue'
+import StrategyPicker from '@/pages/settings/persistence/strategy-picker.vue'
+import ConfigurationPopup from '@/pages/settings/persistence/configuration-popup.vue'
+import FilterPopup from '@/pages/settings/persistence/filter-popup.vue'
+import cloneDeep from 'lodash/cloneDeep'
+import fastDeepEqual from 'fast-deep-equal/es6'
+
+const filterInvertedParameter = {
+  advanced: false,
+  description: 'Whether to invert the above filter, i.e. persist values that do not equal the above values or are outside of the specified range',
+  label: 'Inverted',
+  name: 'inverted',
+  required: false,
+  type: 'BOOLEAN'
+}
+
+export default {
+  mixins: [DirtyMixin],
+  components: {
+    StrategyPicker,
+    'editor': () => import(/* webpackChunkName: "script-editor" */ '@/components/config/controls/script-editor.vue')
+  },
+  props: ['serviceId'],
+  data () {
+    return {
+      newPersistence: false,
+      persistence: {},
+      savedPersistence: {},
+      persistenceYaml: '',
+      ready: false,
+      loading: false,
+      currentTab: 'design',
+      currentConfiguration: null,
+      currentCronStrategy: null,
+      currentFilter: null,
+
+      predefinedStrategies: ['everyChange', 'everyUpdate', 'restoreOnStartup'],
+      filterTypes: [
+        {
+          name: 'thresholdFilters',
```

Comment:
maybe `thresholdFilter`? It's the description of a single filter. Also below.

Reply:
This doesn't work, because the filter type name and label are used by the UI to render the sections and determine where to read from and where to store filters. And the property on the persistence is named `thresholdFilters`.

I have implemented the UI part for filters in a way, that once additional filters are added you only need to update the filterTypes definition and the code tab.